apps/avatax/src/modules/avatax/calculate-taxes/avatax-calculate-taxes-tax-code-matcher.ts

```diff
@@ -14,14 +23,31 @@ export class AvataxCalculateTaxesTaxCodeMatcher {
     if (line.sourceLine.__typename === "OrderLine") {
       return line.sourceLine.orderProductVariant?.product.taxClass?.id;
     }
+
+    throw new AvataxCalculateTaxesTaxCodeMatcherError("Unsupported line type", {
+      props: {
+        // @ts-expect-error: not handled typename is not typed in GraphQL
+        type: line.sourceLine.__typename,
```

Comment:
is it possible? Doesn't graphql guarantee us proper line type?

Reply:
According to TypeScript `sourceLine` in such case will be `never` but this error should be rare edge case - we add new type of `sourceLine` but app didn't yet handle it